shell/server/grain-server.js

```diff
@@ -121,10 +121,34 @@ Meteor.publish("tokenInfo", function (token) {
         let identity = globalDb.getIdentity(apiToken.owner.user.identityId);
         let metadata = apiToken.owner.user.denormalizedGrainMetadata;
         if (identity && metadata) {
+          const login = {};
+          // TODO(cleanup): This switching logic does not belong here. Perhaps it should go
+          //   in sandstorm-db/profile.js or be added as a method on Accounts.identityServices[].
+          if (identity.profile.service === "google") {
+            login.oneClick = { method: "loginWithGoogle",
```

Comment:
I've addressed this by adding the methods `getLoginId()` and `initiateLogin()` to the objects in `Accounts.identityServices`.